File: Examenes/ExamenPSP/2DAM_PSP-main/9 - Caja fuerte asimétrica/CajaFuerte/cliente/src/main/java/ui/screens/register/RegisterController.java
package ui.screens.register;

import io.github.palexdev.materialfx.controls.MFXTextField;
import jakarta.inject.Inject;
import javafx.application.Platform;
import javafx.fxml.FXML;
import ui.screens.common.BaseScreenController;

public class RegisterController extends BaseScreenController {

    private final RegisterViewModel registerViewModel;

    @FXML
    private MFXTextField mfxName;
    @FXML
    private MFXTextField mfxPassword;

    @Inject
    public RegisterController(RegisterViewModel registerViewModel) {
        this.registerViewModel = registerViewModel;
    }

    @Override
    public void principalCargado() {
        registerViewModel.getState().addListener((observableValue, registerState, newValue) -> Platform.runLater(() -> {
                    if (newValue.getError() != null) {
                        getPrincipalController().showAlertError(newValue.getError());

                    }
                    if (newValue.getSuccess() != null) {
                        getPrincipalController().showAlertInformation(newValue.getSuccess());
                        mfxName.setText("");
                        mfxPassword.setText("");
                    }
                    registerViewModel.clearMessages();
                })
        );
    }

    public void addUser() {
        registerViewModel.register(mfxName.getText(), mfxPassword.getText());
    }

    public void backToLogin() {
        getPrincipalController().backToLogin();
    }
}

File: Examenes/ExamenPSP/2DAM_PSP-main/11 - Spring GraphQL/graphqlCliente/src/main/kotlin/ui/controllers/RegisterController.kt
package ui.controllers

import dao.PersonaDao
import io.github.palexdev.materialfx.controls.MFXTextField
import javafx.fxml.FXML
import kotlinx.coroutines.DelicateCoroutinesApi
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.javafx.JavaFx
import kotlinx.coroutines.launch
import ui.common.BaseScreenController
import utils.Trither

class RegisterController: BaseScreenController(){
    @FXML lateinit var mfxSurname: MFXTextField

    @FXML lateinit var mfxName: MFXTextField

    @FXML
    private lateinit var mfxUsername: MFXTextField

    @FXML
    private lateinit var mfxPassword: MFXTextField

    private val dao = PersonaDao

    @OptIn(DelicateCoroutinesApi::class)
    fun addUser() {
        GlobalScope.launch(Dispatchers.JavaFx) {
            dao.register(mfxName.text, mfxSurname.text, mfxUsername.text, mfxPassword.text).collect {
                when (it) {
                    is Trither.Success -> {
                        getPrincipalController()!!.showLoading(false)
                        getPrincipalController()!!.showAlertInformation("Usuario registrado con éxito")
                        backToLogin()
                    }

                    is Trither.Error -> {
                        getPrincipalController()!!.showLoading(false)
                        getPrincipalController()!!.showAlertError(it.message)

                    }

                    else -> {
                        getPrincipalController()!!.showLoading(true)
                    }
                }
            }
        }
    }

    fun backToLogin() {
        getPrincipalController()!!.backToLogin()
    }
}